### PR TITLE
feat(timepicker): kbd and mouse events

### LINKED ISSUE #459: timepicker: add support for arrowkeys and mousewheel
Ref: http://angular-ui.github.io/bootstrap/#/timepicker

It should be trivial to implement but needs tests (as always!)

### PR DESCRIPTION
**Pull request by sendilkumarn**
fix #459

### CODE REVIEW

File: src/timepicker/timepicker.ts [hunk 1]
```diff
@@ -63,17 +63,23 @@ const NGB_TIMEPICKER_VALUE_ACCESSOR = {
       <tr>
         <td>
           <input type="text" class="form-control" maxlength="2" size="2" 
-            [ngModel]="formatHour(model?.hour)" (ngModelChange)="updateHour($event)">
+            [ngModel]="formatHour(model?.hour)" (ngModelChange)="updateHour($event)"
+            (keyup.arrowUp)="changeHour(hourStep)" (keyup.arrowDown)="changeHour(-hourStep)"
+            (mouseup)="changeHour(hourStep)" (mousedown)="changeHour(-hourStep)">
```
Comment (pkozlowski-opensource) [inline src/timepicker/timepicker.ts]: `mouseup` / `mousedown` are _not_ good events! We should be responding to mouse wheel event, not button clicks! See how it is done in angular-ui/bootstrap: https://github.com/angular-ui/bootstrap/blob/c61d16a9330c18105d074ffd23f14db06689809b/src/timepicker/timepicker.js#L207-L238

Applies to all the remaining fields (hours, minutes, seconds)

File: src/timepicker/timepicker.ts [hunk 2]
```diff
@@ -63,17 +63,23 @@ const NGB_TIMEPICKER_VALUE_ACCESSOR = {
       <tr>
         <td>
           <input type="text" class="form-control" maxlength="2" size="2" 
-            [ngModel]="formatHour(model?.hour)" (ngModelChange)="updateHour($event)">
+            [ngModel]="formatHour(model?.hour)" (ngModelChange)="updateHour($event)"
+            (keyup.arrowUp)="changeHour(hourStep)" (keyup.arrowDown)="changeHour(-hourStep)"
```
Comment (pkozlowski-opensource) [inline src/timepicker/timepicker.ts]: I'm not sure `keyup` is the best event as we can't prevent default on it. IMO we should use `keydown` _and_ prevent default.

**Comment by pkozlowski-opensource** [comment]
@sendilkumarn left some comments. This one needs more work before it can be merged.

**Comment by sendilkumarn** [comment]
@pkozlowski-opensource thanks. addressed them. Let me know if you have any.

**Comment by pkozlowski-opensource** [comment]
@sendilkumarn thnx for the update. I've just tested it once again and there are few more items to take care of:
* [ ] - it doesn't merge cleanly, needs rebasing
* [ ] - we need to `preventDefault()` on both mouse scroll events and keyboard up / down events
* [ ] - scroll event should take effect only if a given input element is focused

**Comment by sendilkumarn** [comment]
> preventDefault() on both mouse scroll events and keyboard up / down events

for keyboard, can we catch keydown event and then do something like mouse scroll events? 

I feel it would be ugly to have 2 different methods / using 2 different args

**Comment by sendilkumarn** [comment]
Also can we use 'wheel' instead of 'mousewheel'?

Think mousewheel is not standard anymore

**Comment by sendilkumarn** [comment]
Is this fine ?

**Comment by sendilkumarn** [comment]
@pkozlowski-opensource Is there something missed ?

**Comment by pkozlowski-opensource** [comment]
>  Is there something missed ?

@sendilkumarn sorry, I'm focusing on migration to rc5 atm so haven't had much time for other tasks... I will go back to your PR as soon as I'm done with RC5. Sorry again, there are only 24h / day...

**Comment by sendilkumarn** [comment]
Will close this too @pkozlowski-opensource Thanks :)

**Comment by thbt** [comment]
Why has this PR been closed and what does it need to be merged ?

**Comment by sendilkumarn** [comment]
It was open for a very long time so closed it.
We can reopen and work on top of it

**Comment by thbt** [comment]
Please do that, I would like to make this component accessible and your PR would be a good start!

**Comment by pkozlowski-opensource** [comment]
Closing in favor of #1334 since it contains a cleaned up version of this PR.

**Pull request closed without merging**